programs/rodeos/streams/rabbitmq.hpp

```diff
@@ -72,17 +72,18 @@ class rabbitmq : public stream_handler {
       });
    }
 
-   void declare_exchange(std::string& exchange_type) {
+   void declare_exchange(const std::string& exchange_type) {
       auto type = AMQP::direct;
       if (exchange_type == "fanout") {
          type = AMQP::fanout;
+      } else if (exchange_type == "topic") {
+         type = AMQP::topic;
+      } else {
+         throw std::runtime_error("Unsupported RabbitMQ exchange type: " + exchange_type);
       }
 
       auto& exchange = channel_->declareExchange(exchangeName_, type);
-      exchange.onSuccess([this]() {
-         ilog("RabbitMQ Connected Successfully!\n Exchange ${e}",
-              ("e", exchangeName_));
-      });
+      exchange.onSuccess([this]() { ilog("RabbitMQ Connected Successfully!\n Exchange ${e}", ("e", exchangeName_)); });
```

Comment:
This still has the problem that `this` might be destroyed before this is called. Copy the `exchangeName_` into the lambda.

Reply:
That's what I did first, because I thought that was the cause of SEGFAULT, but then I saw that the lambda is contained in the channel_ data member, so I put `this` back. Because this->exchangeName_ is not in the immediate scope of the lambda, I'd have to use capture with an initializer, which is c++14. Should I do that?

Reply:
I made the requested change.